### PR TITLE
add --private-dns-zone for AKS private cluster

### PR DESCRIPTION
**Pull request by levimm**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
aks

**Comment by yonzhan** [comment]
aks

**Comment by levimm** [comment]
wait for official sdk vendor. This is a test pr with local vendor.

**Review by bcho — APPROVED** [review]

**Review by bingosummer — APPROVED** [review]

File: src/aks-preview/azext_aks_preview/_help.py [hunk 1]
```diff
@@ -242,6 +242,10 @@
         - name: --enable-private-cluster
           type: string
           short-summary: Enable private cluster.
+        - name: --private-dns-zone
+          type: string
+          short-summary: (PREVIEW) private dns zone mode for private cluster.
```
Comment (arrownj) [inline src/aks-preview/azext_aks_preview/_help.py:247]: Please use `is_preview` attribute to make the parameter as Preview.

Comment (levimm) [reply src/aks-preview/azext_aks_preview/_help.py:247]: seems I couldn't find such example for the parameter description here.

Comment (feiskyer) [reply src/aks-preview/azext_aks_preview/_help.py:247]: `is_preview` would should help in a different format compared to other features. prefer `(PREVIEW)` to keep consistency.

**Review by feiskyer — APPROVED** [review]

**Review by zqingqing1 — APPROVED** [review]
lgtm

**Comment by levimm** [comment]
@arrownj  could you help merge the pr?

**Review by arrownj — APPROVED** [review]

**Pull request merged**